src/ArrayManager.inl

```diff
@@ -137,7 +137,9 @@ void* ArrayManager::reallocate(void* pointer, size_t elems)
       pointer_record->m_user_callback(ACTION_ALLOC, ExecutionSpace(space), sizeof(T) * elems);
       void* new_ptr = m_allocators[space]->allocate(sizeof(T)*elems);
 
-      rm.copy(old_ptr, new_ptr);
+      if (pointer_record->m_touched[space]) {
+        rm.copy(old_ptr, new_ptr);
```

Comment:
I assume the rm knows the lengths and this does a reasonable thing here?

Reply:
Yes sir.